app/domain/authentication/authn_jwt/signing_key/fetch_jwks_uri_signing_key.rb

```diff
@@ -0,0 +1,87 @@
+require 'uri'
+require 'net/http'
+require 'base64'
+
+module Authentication
+  module AuthnJwt
+    module SigningKey
+      # This class is responsible for fetching JWK Set from JWKS-uri
+      class FetchJwksUriSigningKey < FetchSigningKeyInterface
+
+        def initialize(
+          authentication_parameters:,
+          fetch_required_secrets: Conjur::FetchRequiredSecrets.new,
+          resource_class: ::Resource,
+          http_lib: Net::HTTP,
+          create_jwks_from_http_response: CreateJwksFromHttpResponse.new,
+          logger: Rails.logger
+        )
+          @logger = logger
+
+          @fetch_required_secrets = fetch_required_secrets
+          @resource_class = resource_class
+          @resource_id = authentication_parameters.authn_jwt_variable_id_prefix
```

Comment:
this should be `@variable_id`

Reply:
also - please separate between dependencies and inputs. do this for all regular classes.

Reply:
Done